resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/submit/submitsteps/initcontainer/SubmittedResourcesInitContainerStepSuite.scala

```diff
@@ -0,0 +1,123 @@
+
+import java.io.File
+import java.util.UUID
+
+import com.google.common.base.Charsets
+import com.google.common.io.{BaseEncoding, Files}
+import io.fabric8.kubernetes.api.model._
+
+import scala.collection.JavaConverters._
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.deploy.kubernetes.InitContainerResourceStagingServerSecretPlugin
+import org.apache.spark.deploy.kubernetes.config._
+import org.apache.spark.deploy.kubernetes.constants._
+import org.apache.spark.deploy.kubernetes.submit.{KubernetesFileUtils, SubmittedDependencyUploader, SubmittedResourceIdAndSecret}
+import org.apache.spark.util.Utils
+import org.mockito.{Mock, MockitoAnnotations}
+import org.mockito.Matchers.{any, eq => mockitoEq}
+import org.mockito.Mockito.when
+import org.scalatest.BeforeAndAfter
+
+class SubmittedResourcesInitContainerStepSuite extends SparkFunSuite with BeforeAndAfter {
+  private val RESOURCE_SECRET_NAME = "secret"
+  private val JARS_RESOURCE_ID = "jarsID"
+  private val JARS_SECRET = "jarsSecret"
+  private val FILES_RESOURCE_ID = "filesID"
+  private val FILES_SECRET = "filesSecret"
+  private def createRSecret = {
+    val jarsSecretBase64 = BaseEncoding.base64().encode(JARS_SECRET.getBytes(Charsets.UTF_8))
+    val filesSecretBase64 = BaseEncoding.base64().encode(FILES_SECRET.getBytes(Charsets.UTF_8))
+    Map(
+      INIT_CONTAINER_SUBMITTED_JARS_SECRET_KEY -> jarsSecretBase64,
+      INIT_CONTAINER_SUBMITTED_FILES_SECRET_KEY -> filesSecretBase64).asJava
+  }
+  private val STAGING_SERVER_URI = "http://localhost:8000"
+  private val SECRET_MOUNT_PATH = "/tmp"
+  private val RSS_SECRET = createRSecret
+
+  @Mock
+  private var submittedDependencyUploader: SubmittedDependencyUploader = _
+  @Mock
+  private var submittedResourcesSecretPlugin: InitContainerResourceStagingServerSecretPlugin = _
+
+  before {
+    MockitoAnnotations.initMocks(this)
+    when(submittedDependencyUploader.uploadJars()).thenReturn(
+      SubmittedResourceIdAndSecret(JARS_RESOURCE_ID, JARS_SECRET)
+    )
+    when(submittedDependencyUploader.uploadFiles()).thenReturn(
+      SubmittedResourceIdAndSecret(FILES_RESOURCE_ID, FILES_SECRET)
+    )
+    when(submittedResourcesSecretPlugin.mountResourceStagingServerSecretIntoInitContainer(
+      mockitoEq(new Container()))).thenReturn(
+        new ContainerBuilder().withName("mountedSecret").build())
+    when(submittedResourcesSecretPlugin.addResourceStagingServerSecretVolumeToPod(
+      mockitoEq(new Pod()))).thenReturn(
+      new PodBuilder()
+        .withNewMetadata()
+        .addToLabels("mountedSecret", "true")
+        .endMetadata()
+        .withNewSpec().endSpec()
+        .build())
+  }
+  test ("testing prepareInitContainer on resources and properties") {
+    val submittedResourceStep = new SubmittedResourcesInitContainerStep(
+      RESOURCE_SECRET_NAME,
+      STAGING_SERVER_URI,
+      SECRET_MOUNT_PATH,
+      false,
+      None,
```

Comment:
Probably worth setting these to `Some(...)` so that we inspect that they're set properly in the resolved Spark configuration. Or, write a separate test for specifically checking how we mount the secrets for some of these fields.